### PR TITLE
Document short-names and categories

### PR DESCRIPTION
**Pull request by liggitt**
xref https://github.com/kubernetes/kubernetes/pull/117742#issuecomment-1548147525

cc @kubernetes/api-reviewers 
/cc @thockin @smarterclayton

### CODE REVIEW

File: contributors/devel/sig-architecture/api-conventions.md [hunk 1]
```diff
@@ -716,6 +719,74 @@ saved. For more details on how to use Merge Patch, see the RFC.
...
+which clients may use as hints when resolving ambiguous user invocations.
+
+For compiled-in resources, these are controlled by the REST handler `ShortNames() []string` and `Categories() []string` implementations.
+
+For custom resources, these are controlled by the `.spec.names.shortNames` and `.spec.names.categories` fields in the CustomResourceDefinition.
+
+### Short-names
+
+Note: Due to unpredictable behavior when short names collide (with each other or with resource types),
+adding new short names to built-in resources is not currently recommended. See issues
```
Comment (thockin) [inline contributors/devel/sig-architecture/api-conventions.md:735]: You are polite.  I'd say "is forbidden unless specifically allowed by API reviewers" .

**Review by thockin — COMMENTED** [review]
/lgtm
/approve
/hold if you want to make words stronger

File: contributors/devel/sig-architecture/api-conventions.md (hunk 1 above)
Comment (liggitt) [reply contributors/devel/sig-architecture/api-conventions.md]: strengthened wording

**Pull request merged**